sdk/face/azure-ai-vision-face/azure/ai/vision/face/_patch.py

```diff
@@ -0,0 +1,975 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from typing import Any, IO, List, Optional, Union, cast
+import sys
+
+from azure.core.tracing.decorator import distributed_trace
+
+from . import models as _models
+from ._client import FaceClient as FaceClientGenerated
+from ._client import FaceSessionClient as FaceSessionClientGenerated
+from ._operations._operations import (
+    FaceClientOperationsMixin,
+    FaceSessionClientOperationsMixin,
+)
+
+if sys.version_info >= (3, 9):
+    from collections.abc import MutableMapping
+else:
+    from typing import MutableMapping  # type: ignore  # pylint: disable=ungrouped-imports
+JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
+
+
+class FaceClient(FaceClientGenerated):
+    """FaceClient.
+
+    :param endpoint: Supported Cognitive Services endpoints (protocol and hostname, for example:
+     https://{resource-name}.cognitiveservices.azure.com). Required.
+    :type endpoint: str
+    :param credential: Credential used to authenticate requests to the service. Is either a
+     AzureKeyCredential type or a TokenCredential type. Required.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or
+     ~azure.core.credentials.TokenCredential
+    :keyword api_version: API Version. Default value is "v1.1-preview.1". Note that overriding this
+     default value may result in unsupported behavior.
+    :paramtype api_version: str or ~azure.ai.vision.face.models.Versions
+    """
+
+    @distributed_trace
+    def detect_from_url(
+        self,
+        url: str,
+        detection_model: Union[str, _models.FaceDetectionModel],
+        recognition_model: Union[str, _models.FaceRecognitionModel],
+        return_face_id: bool,
```

Comment:
Based on the earilest API designs, these three parameters are optional and their default value are "detection_01", "recognition_01" and "true". (`return_face_id` = true).
Nowadays, we have developed and released many more powerful detection and recognition models, and we'd like to push these new models and get more customers to use them. However, we couldn't change the API spec, so many users are still using old model. Therefore, we hope that by making these parameters required, more customers will notice them and try our new model.
For `return_face_id`, due to the Responsible AI principles, customer are not be able to get the response with `return_face_id` as True if they haven't gotten an approval. Thus, we set `return_face_id` as a required parameter to let customers notice it. If this input isn't a bool object, we'll set it as False to the backend by default. (See Line 430.)

Reply:
Got it. Instead of making these required positional, I suggest changing them to required keyword-only, like:

```python
    def detect_from_url(
        self,
        *,
        url: str,
        detection_model: Union[str, _models.FaceDetectionModel],
        recognition_model: Union[str, _models.FaceRecognitionModel],
        return_face_id: bool,
        return_face_attributes: Optional[List[Union[str, _models.FaceAttributeType]]] = None,
        return_face_landmarks: Optional[bool] = None,
        return_recognition_model: Optional[bool] = None,
        face_id_time_to_live: Optional[int] = None,
        **kwargs: Any,
    ) -> List[_models.FaceDetectionResult]: ...
```

There's a few benefits with this - if you ever decide to make these parameters optional again, it won't be breaking. Also, we can keep the overloads in this case, so that if someone would rather pass JSON or IO[bytes] they can.

Reply:
Great suggestion. The interface of `detect_from_url` and `detect` now look like this:

```python
    # It allows passing JSON or IO[bytes] as the `body` parameter if the users wish.
    # `url` is required if the body is empty.
    def detect_from_url(
        self,
        body: Union[JSON, IO[bytes]] = _Unset,
        *,
        url: str = _Unset,
        detection_model: Union[str, _models.FaceDetectionModel],
        recognition_model: Union[str, _models.FaceRecognitionModel],
        return_face_id: bool,
        return_face_attributes: Optional[List[Union[str, _models.FaceAttributeType]]] = None,
        return_face_landmarks: Optional[bool] = None,
        return_recognition_model: Optional[bool] = None,
        face_id_time_to_live: Optional[int] = None,
        **kwargs: Any,
    ) -> List[_models._models.FaceDetectionResult]:
```

```python
    def detect(
        self,
        image_content: bytes,
        *,
        detection_model: Union[str, _models.FaceDetectionModel],
        recognition_model: Union[str, _models.FaceRecognitionModel],
        return_face_id: bool,
        return_face_attributes: Optional[List[Union[str, _models.FaceAttributeType]]] = None,
        return_face_landmarks: Optional[bool] = None,
        return_recognition_model: Optional[bool] = None,
        face_id_time_to_live: Optional[int] = None,
        **kwargs: Any,
    ) -> List[_models._models.FaceDetectionResult]:
```
See the commit: https://github.com/Azure/azure-sdk-for-python/pull/35384/commits/5ee2b8bb14f205039cb4092d3a9730b9e401b3ab and https://github.com/Azure/azure-sdk-for-python/pull/35384/commits/0623a38f98c92ba7e7133b57333e273cf8be7e95.